hudi-timeline-service/src/main/java/org/apache/hudi/timeline/service/TimelineService.java

```diff
@@ -98,16 +98,42 @@ public TimelineService(Config config) throws IOException {
     public Boolean help = false;
   }
 
+  private int startServiceOnPort(int port) throws IOException {
+    if (!(port == 0 || (1024 <= port && port < 65536))) {
+      throw new IllegalArgumentException(String.format("startPort should be between 1024 and 65535 (inclusive), "
+          + "or 0 for a random free port. but now is %s.", port));
+    }
+    int maxRetries = 16;
+    for (int offset = 0; offset < maxRetries; offset++) {
+      int tryPort = port == 0 ? port : (port + offset - 1024) % (65536 - 1024) + 1024;
```

Comment:
what are we trying to do in the `:` else part? a line of comment could help

Reply:
done